pkg/observability/tracing/tracing.go

```diff
@@ -65,3 +71,25 @@ func registerTraceExporter(res *resource.Resource, exporter sdktrace.SpanExporte
 	// Shutdown will flush any remaining spans and shut down the exporter.
 	return tp.Shutdown, nil
 }
+
+func Start(ctx context.Context, name string) (context.Context, trace.Span) {
+	return otel.Tracer(tracerName).Start(ctx, name)
+}
+
+// Error returns an error representing code and error message and records new event on the span.
+// If code is OK, returns nil.
+func Error(span trace.Span, code grpccodes.Code, err error, events ...string) error {
+	for _, event := range events {
+		span.AddEvent(event)
```

Comment:
Would it make sense to check if span is nil before doing anything else? You're calling a method on span before the nil check so it could go bang